docs/sensors.md

```diff
@@ -0,0 +1,158 @@
+ZJS API for W3C Generic Sensors
+==============================
+
+* [Introduction](#introduction)
+* [Web IDL](#web-idl)
+* [API Documentation](#api-documentation)
+* [Sample Apps](#sample-apps)
+
+Introduction
+------------
+ZJS Generic Sensor API implements the W3C Sensor API, and it's intended to
+provide a consistent API that allow apps to communicate with sensors like
+accelerometer and gyroscope. Since the W3C Sensor API is still a draft spec,
+our implementation only provide a subset of the API and the API could be
+slightly different, but we try to follow the latest spec as close as possible.
+
+The currently supported hardware are Arduino 101s with its built-in BMI160
+accelerometer and gyroscope, also ambient light sensor that can be connected
+using analog pin.
+
+Web IDL
+-------
+This IDL provides an overview of the interface; see below for documentation of
+specific API functions.
+
+Sensor Interface
+```javascript
+[SecureContext]
+interface Sensor : EventTarget {
+  readonly attribute SensorState state; // The current state of Sensor object
+  void start();                         // Starts the sensor
+  void stop();                          // Stops the sensor
+  attribute EventHandler onchange;      // Event handler for change events
+  attribute EventHandler onactivate;    // Event handler for activate events
+  attribute EventHandler onerror;       // Event handler for error events
+};
+
+dictionary SensorOptions {
+  double? frequency;                    // The requested polling frequency, default is 20 if unset
+};
+
+enum SensorState {
+  "unconnected",
+  "activating",
+  "activated",
+  "idle",
+  "errored"
+};
+
+[SecureContext, Constructor(DOMString type, SensorErrorEventInit errorEventInitDict)]
+interface SensorErrorEvent : Event {
+  readonly attribute Error error;
+};
+
+dictionary SensorErrorEventInit : EventInit {
+  required Error error;
+};
+```
+AccelerometerSensor Interface
+```javascript
+[Constructor(optional AccelerometerSensorOptions accelerometerSensorOptions)]
+interface AccelerometerSensor : Sensor {
+  readonly attribute AccelerometerSensorReading? reading;
```

Comment:
Question mark means the value can be null, and "docs should describe when this can happen" according to some web page I just found.

So check all your usage of ? and make sure this is true and if so documented based on our actual implementation.